internal/strace/strace.go

```diff
@@ -180,19 +174,19 @@ func (r *Result) parseEnterSyscall(syscall, args string) error {
 		match := writePattern.FindStringSubmatch(args)
 		// Example: django.template.base\nImporting django.template"..., 0xe64
 		writeBufAndBytesWritten := match[2]
-		for i := len(writeBufAndBytesWritten) - 1; i >= 0; i-- {
-			// TODO: A future PR will use the index right before the comma to get the end of the write buffer.
-			if writeBufAndBytesWritten[i] == ',' {
-				// Use the index after the space and "0x" to get to the first index of the bytes written.
-				bytesWritten, err := strconv.ParseInt(writeBufAndBytesWritten[i+4:len(writeBufAndBytesWritten)], 16, 64)
-				r.recordFileWrite(match[1], bytesWritten)
-				if err != nil {
-					return err
-				}
-				break
-			}
+		// The index of the comma that separates the write buffer argument from the bytes written argument.
+		commaIndex := strings.LastIndex(writeBufAndBytesWritten, ",")
+		if commaIndex == -1 || len(writeBufAndBytesWritten) <= commaIndex+4 || writeBufAndBytesWritten[commaIndex+1:commaIndex+4] != " 0x" {
```

Comment:
a comment explaining where the +4 comes from (or storing it in a helpfully named variable) would be good for readers

Reply:
+1 ! This is also getting complicated enough that it may be worth putting this in a separate function.

Reply:
I simplified the parsing logic for bytes written. I'll parse the file contents separately when storing them. This will also address this issue: https://github.com/ossf/package-analysis/issues/477

Reply:
It's now only one or statement. I can pull it out to a separate function if the logic gets more complicated when we add in parsing for file contents though